src/tests/rust_fmt_argument_max_padding.rs

```diff
@@ -0,0 +1,244 @@
+//! These tests exist to validate padding behavior when using dynamic widths in
+//! format strings.  As of rust-lang/rust#136932 (part of #99012), the Rust
+//! standard library now restricts width and precision fields in format strings
+//! to `u16::MAX`, to improve memory layout and prevent silent truncation bugs
+//! on cross-compilation targets. This change caused previously valid dynamic
+//! width values exceeding `u16::MAX` to panic at runtime. (See
+//! rust-lang/rust#136932 and artichoke/strftime-ruby#2f0ab4a).
+//!
+//! These test cases specifically target edge conditions revealed by fuzzing
+//! strftime-ruby, ensuring we handle excessively large padding values without
+//! panicking, aligning behavior closely with CRuby's `Time#strftime`
+//! specification and the limitations described in rust-lang/rust#136932.
+//!
+//! Reference:
+//!
+//! - <https://github.com/rust-lang/rust/pull/136932>
+//! - <https://github.com/rust-lang/rust/pull/136932#issuecomment-2739434542>
+
+#![allow(clippy::unreadable_literal)]
+
+#[cfg(feature = "alloc")]
+use {
+    crate::format::TimeFormatter,
+    alloc::{format, vec},
+};
+
+use crate::Error;
+
+use super::{check_all, get_format_err, get_format_err_bytes, MockTime};
+
+#[test]
+fn test_larger_than_int_max_formats_are_returned_verbatim() {
+    let times = [
+        MockTime::new(1970, 1, 1, 0, 0, 0, 0, 4, 1, 0, false, 0, ""),
+        MockTime::new(-1970, 1, 1, 0, 0, 0, 0, 4, 1, 0, false, 0, ""),
+    ];
+
+    for format in [
+        "%100000000000000000000c",
+        "%1000000000000c",
+        "%10000000000c",
+    ] {
+        check_all(&times, format, &[format, format]);
+    }
+}
+
+#[test]
+fn test_rust_136932_reduce_fmt_argument_width_and_precision_fuzzer_failures() {
```

Comment:
I don't think these tests are very useful, since they are pretty difficult to understand and are covered by the tests below.

Reply:
these came straight from the fuzzer failures, so this is what I started with while debugging. I figure it can't hurt to keep them, but if you feel strongly, I can remove

Reply:
We identified the root cause, so it is better to only keep the new tests since they are exhaustive are more targeted.

Reply:
sounds good